### PR TITLE
SK-568: Telegram credential

### PR DESCRIPTION
**Pull request by arty-name**
SK-568

### CODE REVIEW

File: src/frontend/views/Telegram/Telegram.module.css [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+.buttonsLine {
+  text-align: center;
+}
+
+.iframe {
+  overflow: hidden;
+  border: medium none;
+  height: 2rem;
+  width: 16rem;
+}
```
Comment (lukeg90) [inline src/frontend/views/Telegram/Telegram.module.css:15]: Can we center the contents of the iframe somehow?

File: src/backend/telegram/confirmTelegram.ts [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+  const computed = createHmac('sha256', secret)
+    .update(plaintext)
+    .digest()
+    .toString('hex');
+
+  if (hash !== computed) {
+    throw Boom.forbidden('Auth data hash doesn’t match');
+  }
+}
```
Comment (lukeg90) [inline src/backend/telegram/confirmTelegram.ts:62]: Maybe someone who understands crypto better than me should review this function 🥴

**Review by lukeg90 — APPROVED** [review]

File: src/frontend/views/Telegram/Telegram.module.css (hunk 1 above)
Comment (arty-name) [reply src/frontend/views/Telegram/Telegram.module.css:15]: Nope, the internet security prevents us from doing that

File: src/backend/telegram/confirmTelegram.ts (hunk 2 above)
Comment (arty-name) [reply src/backend/telegram/confirmTelegram.ts:62]: This is described here https://core.telegram.org/widgets/login#checking-authorization, and the check passes, so the hashes are calculated correctly. But we could ask someone else for your peace of mind :)

Comment (lukeg90) [reply src/backend/telegram/confirmTelegram.ts:62]: Oh I see. My mind is at peace. Go ahead and merge it 😄

Comment (rflechtner) [reply src/backend/telegram/confirmTelegram.ts:62]: Yeah looks good to me too

Comment (arty-name) [reply src/backend/telegram/confirmTelegram.ts:62]: Thank you Raphael!

**Pull request merged**